metricflow/test/model/validations/helpers.py

```diff
@@ -0,0 +1,97 @@
+from typing import List, Optional, Sequence
+
+from metricflow.dataflow.sql_table import SqlTable
+from metricflow.engine.models import Dimension
+from metricflow.model.objects.common import FileSlice, Metadata
+from metricflow.model.objects.constraints.where import WhereClauseConstraint
+from metricflow.model.objects.data_source import DataSource, DataSourceOrigin, Mutability
+from metricflow.model.objects.elements.identifier import Identifier
+from metricflow.model.objects.elements.measure import Measure
+from metricflow.model.objects.materialization import Materialization, MaterializationDestination
+from metricflow.model.objects.metric import Metric, MetricType, MetricTypeParams
+
+
+def default_meta() -> Metadata:
+    """Returns a Metadata object with the required information"""
+
+    return Metadata(
+        repo_file_path="/not/from/a/repo",
+        file_slice=FileSlice(
+            filename="not_from_file.py",
+            content="N/A",
+            start_line_number=0,
+            end_line_number=0,
+        ),
+    )
+
+
+def materialization_with_guaranteed_meta(
+    name: str,
+    metrics: List[str],
+    dimensions: List[str],
+    description: str = "adhoc materialization",
+    metadata: Metadata = default_meta(),
+    destinations: List[MaterializationDestination] = [],
+    destination_table: Optional[SqlTable] = None,
```

Comment:
I don't love these shadow factory functions but I have no better ideas to offer unless you don't necessarily care about the metadata values themselves.

Reply:
I don't necessarily care about the metadata values themselves, so I would love any ideas 😄